ui/src/timeMachine/components/TagSelector.tsx

```diff
@@ -67,7 +73,17 @@ class TagSelector extends PureComponent<Props> {
   private debouncer = new DefaultDebouncer()
 
   public render() {
-    return <div className="tag-selector">{this.body}</div>
+    const {index} = this.props
+
+    return (
+      <BuilderCard>
+        <BuilderCard.Header
+          title="Filter"
+          onDelete={index !== 0 && this.handleRemoveTagSelector}
```

Comment:
I feel like `index !== 0` should be a guard inside the handler

Reply:
It's definitely funky. If a function is passed in at all the delete button appears on the card, so I threw the guard in here so it doesn't end up with a non-functional delete button. I think long term this will change as cards will be more flexible